src/ess/sans/q1d.py

```diff
@@ -0,0 +1,128 @@
+
+    # Solid angle
+    solid_angle = normalization.solid_angle(data,
+                                            pixel_width=data.coords['pixel_width'],
+                                            pixel_height=data.coords['pixel_height'])
+
+    # Interpolate the direct beam function to the requested wavelength binning
+    interpolate_direct_beam = True
+    if direct_beam.coords['wavelength'].sizes['wavelength'] > direct_beam.sizes[
+            'wavelength']:
+        if sc.identical(direct_beam.coords['wavelength'], wavelength_bins):
+            interpolate_direct_beam = False
+    else:
+        if sc.identical(
+                to_bin_edges(direct_beam.coords['wavelength'], dim='wavelength'),
+                wavelength_bins):
+            interpolate_direct_beam = False
+    if interpolate_direct_beam:
+        func = interp1d(sc.values(direct_beam), 'wavelength')
+        direct_beam = func(wavelength_bins, midpoints=True)
+        logger = sc.get_logger()
+        logger.warning('ess.sans.q1d: An interpolation was performed on the '
+                       'direct_beam function. The variances in the direct_beam '
+                       'function have been ignored.')
+
+    # Compute denominator and copy coords needed for conversion to Q
+    denominator = (solid_angle * direct_beam * monitors_wav['data_incident_monitor'] *
+                   transmission_fraction)
+    denominator.coords['position'] = data.meta['position']
+    denominator.coords['gravity'] = data.meta['gravity']
+    denominator.coords['sample_position'] = data.meta['sample_position']
+    denominator.coords['source_position'] = data.meta['source_position']
+    denominator.coords['wavelength'] = to_bin_centers(denominator.coords['wavelength'],
+                                                      dim='wavelength')
+
+    # 3. Convert from wavelength to Q ==================================================
+
+    wavelength_bands = sc.linspace(dim='wavelength',
+                                   start=wavelength_bins.min().value,
+                                   stop=wavelength_bins.max().value,
+                                   num=number_of_wavelength_bands + 1,
+                                   unit=wavelength_bins.unit)
+
+    data = data.transform_coords("Q", graph=graph)
+    q_boundaries = sc.concat([q_bins.min(), q_bins.max()], dim='Q')
+
+    # TODO: once scipp-0.12 is out, we no longer need to move the attr into the coords
+    data.bins.coords['wavelength'] = data.bins.attrs.pop('wavelength')
+
+    q_binned = sc.bin(data, edges=[wavelength_bands, q_boundaries])
+    q_summed = q_binned.bins.concat('spectrum')
+
+    denominator_bands = []
+    for i in range(number_of_wavelength_bands):
+        band = denominator['wavelength',
+                           wavelength_bands['wavelength',
+                                            i]:wavelength_bands['wavelength', i + 1]]
```

Comment:
Note that you can now write:
```suggestion
        band = denominator['wavelength', wavelength_bands[i]:wavelength_bands[i + 1]]
```